freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -649,13 +655,13 @@ def _add_parent_nvrs(nvrs_mapping: dict[str, str]) -> dict[str, str]:
             latest_rebuild = ArtifactBuild.query.filter_by(rebuilt_nvr=latest_nvr).first()
 
             # Check whether the original NVR is also a freshmaker rebuild
-            build = ArtifactBuild.query.filter_by(rebuilt_nvr=latest_rebuild.original_nvr).first()
-            while build:
-                nvr_mappings_to_add[build.rebuilt_nvr] = latest_nvr
-                build = ArtifactBuild.query.filter_by(rebuilt_nvr=build.original_nvr).first()
-        
-        nvrs_mapping.update(nvr_mappings_to_add)
-        return nvrs_mapping
+            former_build = ArtifactBuild.query.filter_by(rebuilt_nvr=latest_rebuild.original_nvr).first()
+            while former_build:
+                nvr_mappings_to_add[former_build.rebuilt_nvr] = latest_nvr
+                log.info(f"Found former rebuild '{former_build.rebuilt_nvr}' of '{latest_nvr}'")
+                former_build = ArtifactBuild.query.filter_by(rebuilt_nvr=former_build.original_nvr).first()
+
+        return{**nvrs_mapping, **nvr_mappings_to_add}
```

Comment:
I'd suggest:
1. If you want to update the `nvr_mapping` directly, then just update this function to `_add_previous_rebuilds(nvrs_mapping: dict[str, str]) -> None`, and have:
```
while former_build:
    nvr_mapping[former_build.rebuilt_nvr] = latest_nvr
```
2. Or do it in more safe way (though seems not so necessary in this simple function):
```
nvrs_mapping = copy.deepcopy(nvrs_mapping)
...
while former_build:
    nvr_mapping[former_build.rebuilt_nvr] = latest_nvr
    ...
return nvrs_mapping
```

Reply:
I guess I feel more comfortable with explicitly setting nvrs_mapping to a new value, instead of changing it implicitly.
I changed the name of the new function to `_completion_with_previous_rebuilds`, as I thought the latter name was not making that much sense with that way of using. Now we have:
```python
nvrs_mapping = self._completion_with_previous_rebuilds(nvrs_mapping)
```

What do you think?

Reply:
Yes, it looks good to me.